modules/internal/src/main/scala/service.scala

```diff
@@ -184,22 +184,54 @@ object serviceImpl {
       private val Client                        = TypeName("Client")
       val clientClass: ClassDef =
         q"""
-        class $Client[$F_](
-          channel: _root_.io.grpc.Channel,
+        class $Client[F[_]](
+          channel: _root_.io.grpc.ManagedChannel,
           options: _root_.io.grpc.CallOptions = _root_.io.grpc.CallOptions.DEFAULT
         )(implicit
-          F: _root_.cats.effect.ConcurrentEffect[$F],
+          F: _root_.cats.effect.ConcurrentEffect[F],
           EC: _root_.scala.concurrent.ExecutionContext
-        ) extends _root_.io.grpc.stub.AbstractStub[$Client[$F]](channel, options) {
+        ) extends _root_.io.grpc.stub.AbstractStub[$Client[F]](channel, options) {
           override def build(channel: _root_.io.grpc.Channel, options: _root_.io.grpc.CallOptions): $Client[F] =
-              new $Client[$F](channel, options)
+              new $Client[F](channel.asInstanceOf[_root_.io.grpc.ManagedChannel], options)
```

Comment:
Both `Client` class and `build` method has a parameter called `channel` but the type of the method parameter is `Channel` so the casting is required. A workaround could be renaming one of the parameters and use the one provided by the class constructor here.

Does it make sense?

Reply:
I think we don't need to use `ManagedChannel` in the class args since we're not using any of their methods. We use it to call service methods, and these methods receive a `Channel`:
https://github.com/higherkindness/mu/blob/d0e7efa942e6fc3696a27d0f303a0758d42389c2/modules/internal/src/main/scala/client/fs2Calls.scala#L34